include/solvers/petsc_linear_solver.h

```diff
@@ -257,6 +258,16 @@ class PetscLinearSolver : public LinearSolver<T>
    */
   virtual LinearConvergenceReason get_converged_reason() const override;
 
+protected:
+#if defined(LIBMESH_ENABLE_AMR) && defined(LIBMESH_HAVE_METAPHYSICL)
```

Comment:
Do we need these guards?  I thought the wrapper was good for field splits even on an unrefined mesh, and I know MetaPhysicL is only for the refinement+GMG case.